site/content/entry/citra-progress-report-2018-q2.md

```diff
@@ -0,0 +1,305 @@
+camera from where the light source begins. Then, from that rendering the depth
+map is extracted. That is then applied as a texture to the darkened scene, making
+areas of it directly visible from the light source appear more brightly lit.
+
+The PICA200 and OpenGL both implement this in different ways, particularly in
+two important areas:
+
+ * The first is that PICA200 supports a variant called "soft shadows", where the
+depth map is blurred before being applied to the scene. This results in shadows
+that seem less jagged and sharp, making the light source feel more diffuse and
+evenly spread out throughout the scene. OpenGL doesn't support this at all.
+
+ * The second is that the PICA200 stores depth maps intended for shadow mapping as
+plain textures in the RGBA8 format. A lot of games exploit this in order to quickly
+convert other types of textures from an internal format to RGBA8. But, OpenGL
+stores these maps in a format internal to that graphics card, like any other map.
+
+The first point isn't as significant, since the softness can be ignored on OpenGL,
+resulting in a very fast (but inaccurate) shadow mapping. Because games rarely
+use soft shadows, this can be ignored relatively safely.
+
+The second point though, because games use it very often, it has to be implemented
+accurately. The naïve way would be to simply convert internal textures to RGBA8
+manually, but this would slow rendering down to a crawl. The smart way would be
+to try and trick the graphics card driver into performing a similar conversion,
+but this exposed a lot of bugs in it, leading to instability.
+
+After some very hard work by [wwylele](https://github.com/wwylele), both problems
+were solved by one stone. OpenGL supports an extension called Image Load/Store,
+which allows shaders to read and write to any texture directly. Using this, he
+created a shader that could accurately implement both soft shadows, and convert
+the depth map from its internal format to RGBA8 very quickly.
+
+{{< sidebyside "image" "/images/entry/citra-progress-report-2018-q2/"
+    "shadow_map_before.png=Before Shadow Mapping"
+    "shadow_map_after.png=After Shadow Mapping" >}}
+
+Unfortunately, because Image Load/Store is an optional extension, not every
+OpenGL 3.3 graphics card will support it. In these cases, Citra will simply ignore
+shadow maps, making the rendering inaccurate but usable. Image Load/Store became
+a mandatory part of OpenGl in version 4.2, so every graphics card that complies
+with OpenGL 4.2 is guaranteed to work correctly.<br>
+**Citra only requires a minimum
+compliance to OpenGL 3.3, but OpenGL 4.2 compliance may lead to more accurate
+rendering.**
+
+## [Implement cubeb audio backend](https://github.com/citra-emu/citra/pull/3776) by [darkf](https://github.com/darkf)
+
+[`cubeb`](https://github.com/kinetiknz/cubeb/) is an audio library written by
+the Mozilla Foundation that's cross-platform, and *fast*. Reports from [Dolphin's
```

Comment:
Could flow better - e.g.:
`is an audio library (developed by the Mozilla Foundation) that's cross-platform, and most importantly, *fast*.`